torch/sparse/__init__.py

```diff
@@ -81,6 +81,8 @@ def mm(mat1: Tensor, mat2: Tensor) -> Tensor:
                values=tensor([ 0.1394, -0.6415, -2.1639,  0.1394, -0.6415, -2.1639]),
                size=(2, 3), nnz=6, layout=torch.sparse_coo)
     """
+    if mat1.is_sparse and mat2.is_sparse:
+        return torch._sparse_sparse_matmul(mat1, mat2)
```

Comment:
Typically, it will be better to do this test inside `_sparse_mm` itself, so that the newly handled situations can also be seen in C++. However, there is something dodgy going on with `torch.sparse.mm` existing at all in the first place; wondering why this isn't just `torch.mm` (semantic difference?)

Reply:
Or maybe it's a backwards thing, in which case we should have some composite op that switches between the sparse-spare and dense-spare versions

Reply:
Yes I agree, to the best of my knowledge,  there is an undocumented convention that functions in `torch`  namespace assume zero fill-value and functions in `torch.sparse` namespace assume undefined fill-value. This should be clear  once the fill-value support is implemented, the `torch.sparse`  will be used only for functions that do not use the specified fill-value. cc @pearu

Reply:
Oh yes, so sparse.mm has a different backwards definition than mm. Makes sense.